core/src/marker/markerManager.cpp

```diff
@@ -72,6 +72,11 @@ bool MarkerManager::setBitmap(MarkerID markerID, int width, int height, const un
     texture->setData(bitmapData, size);
 
     marker->setTexture(std::move(texture));
+
+    // The geometry is unchanged, but the mesh must be rebuilt because DynamicQuadMesh contains
+    // texture batches as part of its data.
+    buildGeometry(*marker, m_zoom);
```

Comment:
And this whole thing has reinforced to me how convoluted it can be to put a marker on the map. There's gotta be something better we can do :\